api/include/opentelemetry/trace/span_context.h

```diff
@@ -44,33 +43,51 @@ class SpanContext final
       : trace_id_(),
         span_id_(),
         trace_flags_(trace_api::TraceFlags((uint8_t)sampled_flag)),
-        is_remote_(is_remote)
+        is_remote_(is_remote),
+        trace_state_(TraceState::GetDefault())
   {}
 
+  SpanContext(TraceId trace_id,
+              SpanId span_id,
+              TraceFlags trace_flags,
+              bool is_remote,
+              nostd::shared_ptr<TraceState> trace_state = TraceState::GetDefault()) noexcept
+      : trace_id_(trace_id),
+        span_id_(span_id),
+        trace_flags_(trace_flags),
+        is_remote_(is_remote),
+        trace_state_(trace_state)
+  {}
+
+  SpanContext(const SpanContext &ctx) = default;
+
   // @returns whether this context is valid
   bool IsValid() const noexcept { return trace_id_.IsValid() && span_id_.IsValid(); }
 
   // @returns the trace_flags associated with this span_context
   const trace_api::TraceFlags &trace_flags() const noexcept { return trace_flags_; }
 
+  // @returns the trace_id associated with this span_context
   const trace_api::TraceId &trace_id() const noexcept { return trace_id_; }
 
+  // @returns the span_id associated with this span_context
   const trace_api::SpanId &span_id() const noexcept { return span_id_; }
 
-  SpanContext(TraceId trace_id, SpanId span_id, TraceFlags trace_flags, bool is_remote) noexcept
-      : trace_id_(trace_id), span_id_(span_id), trace_flags_(trace_flags), is_remote_(is_remote)
-  {}
-
-  SpanContext(const SpanContext &ctx) = default;
-
-  SpanContext &operator=(const SpanContext &ctx) = default;
+  // @returns the trace_state associated with this span_context
+  const nostd::shared_ptr<trace_api::TraceState> trace_state() const noexcept
+  {
+    return trace_state_;
+  }
 
   bool operator==(const SpanContext &that) const noexcept
   {
+    // TODO: check trace_state?
```

Comment:
Can we remove this TODO, but rather add a doc comment to the operator method that describes the current behavior (that trace state is not taken into account when comparing)? The current behavior makes sense in my opinion.